server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -706,19 +726,79 @@ public Node(final Environment environment,
                 fsHealthService
             );
             this.nodeService = new NodeService(monitorService, indicesService, transportService);
+            DDLClusterStateService ddlClusterStateService = new DDLClusterStateService();
+            RoleManagerService rolesManager = new RoleManagerService(
+                client,
+                roles,
+                ddlClusterStateService,
+                clusterService
+            );
+            AnalysisRegistry analysisRegistry = analysisModule.getAnalysisRegistry();
+            AtomicReference<Injector> injectorRef = new AtomicReference<>();
+            Provider<DependencyCarrier> dependencyCarrier = () -> injectorRef.get().getInstance(DependencyCarrier.class);
```

Comment:
Could you please explain why we need this 2 level of supplier?
(Maybe add a comment as well)

Reply:
Oh this shouldn't even be part of this PR but of https://github.com/crate/crate/pull/16152